featuregates/go.mod

```diff
@@ -0,0 +1,28 @@
+module github.com/openshift/cluster-config-operator/featuregates
```

Comment:
Are we doing this to avoid Go mod hell for other repos because of the many deps pulled in by the main module? Or is this to avoid circular module deps?

If it's the first, I think lets try and stay away from this, in theory Go's dep resolution should be able to pick up only the deps you need for the packages you import from a module, and should be smart enough to not cause issues